File: web/src/utils/urls/post-long-title.ts
import memoize from "lodash/memoize";
import { AccountType } from "src/models/account";
import { Post } from "src/models/post";

export const _getPostLongTitle = (post: Pick<Post, "title">, poster?: AccountType) => {
  const posterExtension = poster
    ? " needed " +
      (poster.type === "Company"
        ? `at ${poster.company_name}`
        : `by ${poster.first_name} ${poster.last_name}`)
    : "";

  return `${post.title}${posterExtension}`;
};

export const getPostLongTitle = memoize(_getPostLongTitle);

File: api/src/_utils/post_long_title.rs
use crate::{
  account::model::{Account, AccountNameTrait, AccountType},
  post::model::Post,
};

pub fn get_post_long_title(post: &Post, poster: &Account) -> String {
  let poster_extension = format!(
    " needed {} {}",
    match poster.r#type {
      AccountType::Company { .. } => "at",
      _ => "by",
    },
    poster.get_display_name()
  );

  format!("{}{}", post.title, poster_extension)
}
